test/acceptance/tests/controller/controller_namespaces_test.go

```diff
@@ -0,0 +1,123 @@
+package controller
+
+import (
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/consul-helm/test/acceptance/framework"
+	"github.com/hashicorp/consul-helm/test/acceptance/helpers"
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/sdk/testutil/retry"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	KubeNS       = "ns1"
+	ConsulDestNS = "from-k8s"
+)
+
+// Test that the controller works with Consul Enterprise namespaces.
+// These tests currently only test non-secure and secure without auto-encrypt installations
+// because in the case of namespaces there isn't a significant distinction in code between auto-encrypt
+// and non-auto-encrypt secure installations, so testing just one is enough.
+func TestControllerNamespaces(t *testing.T) {
+	cfg := suite.Config()
+	if !cfg.EnableEnterprise {
+		t.Skipf("skipping this test because -enable-enterprise is not set")
+	}
+
+	cases := []struct {
+		name                 string
+		destinationNamespace string
+		mirrorK8S            bool
+		secure               bool
+	}{
+		{
+			"single destination namespace (non-default)",
+			ConsulDestNS,
+			false,
+			false,
+		},
+		{
+			"single destination namespace (non-default); secure",
+			ConsulDestNS,
+			false,
+			true,
+		},
+		{
+			"mirror k8s namespaces",
+			KubeNS,
+			true,
+			false,
+		},
+		{
+			"mirror k8s namespaces; secure",
+			KubeNS,
+			true,
+			true,
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			ctx := suite.Environment().DefaultContext(t)
+
+			helmValues := map[string]string{
+				"global.enableConsulNamespaces": "true",
+				// todo: remove
+				"global.imageK8S": "lkysow/consul-k8s-dev:sep09-crd-ent6",
+				// todo: end
```

Comment:
todo: before merging switch `values.yaml` to use the docker image we build on merges to `crd-controller-base` for consul-k8s